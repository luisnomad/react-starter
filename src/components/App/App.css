.container {
  display: flex;
  align-items: center;
  min-height: 24em;
  justify-content: center;
}
