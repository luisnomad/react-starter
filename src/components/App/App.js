import React, { Component, PropTypes } from 'react'
import React from 'react'
import PropTypes from 'prop-types'

const App = ({ message }) => (
  <div>
    {message}
  </div>
)

const { string } = PropTypes
App.propTypes = {
  message: string.isRequired
}

export default App
