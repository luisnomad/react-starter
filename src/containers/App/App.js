import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Message from 'components/Message'
import { getMessage } from 'selectors'
import styles from './App.css'

export const App = ({ message }) => (
  <div className={styles.container}>
    <Message text={message} />
  </div>
)

const { string } = PropTypes
App.propTypes = {
  message: string.isRequired
}

const mapStateToProps = state => ({
  message: getMessage(state)
})

export default connect(mapStateToProps)(App)
