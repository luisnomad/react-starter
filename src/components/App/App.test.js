import React from 'react'
import App from './'
import renderer from 'react-test-renderer'

test('App snapshot', () => {
  const component = renderer.create(<App />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
