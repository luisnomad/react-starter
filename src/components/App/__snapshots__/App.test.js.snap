// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="App"
>
  <span
    className="App--message"
  >
    It Works!
  </span>
</div>
`;
